src/main/java/Tree/BinarySearchTree.java

```diff
@@ -0,0 +1,63 @@
+package Tree;
+
+import java.util.List;
+
+public class BinarySearchTree {
+
+    public static void main(String[] args) {

```

Comment:
You should not have a main method in any of these files. This is a library, not an application.